sdk/servicebus/service-bus/src/session/messageSession.ts

```diff
@@ -257,35 +257,27 @@ export class MessageSession extends LinkEntity<Receiver> {
     }
   }
 
-  protected createRheaLink(
+  protected async createRheaLink(
     options: ReceiverOptions,
     _abortSignal?: AbortSignalLike
   ): Promise<Receiver> {
-    return this._context.connection.createReceiver(options);
-  }
-
-  /**
-   * Creates a new AMQP receiver under a new AMQP session.
-   */
-  private async _init(
-    opts: { abortSignal?: AbortSignalLike; timeoutInMs?: number } = {}
-  ): Promise<void> {
-    try {
-      const sessionOptions = this._createMessageSessionOptions(this.identifier, opts.timeoutInMs);
-      await this.initLink(sessionOptions, opts.abortSignal);
-
-      if (this.link == null) {
-        throw new Error("INTERNAL ERROR: failed to create receiver but without an error.");
-      }
-
-      const receivedSessionId =
-        this.link.source &&
-        this.link.source.filter &&
-        this.link.source.filter[Constants.sessionFilterName];
+    this._lastSBError = undefined;
+    const link = await this._context.connection.createReceiver(options);
 
+    // When we ask for any sessions, but don't get one back, check whether
+    // service has sent any error.
+    if (
+      options.source &&
```

Comment:
what is the difference between `options.source` and `link.source`?

Reply:
options is input and link is the result that we get back

Reply:
No they are not always the same.  When we pass session filter of `undefined`, we are asking for any available one, successful link would have the session name that service returned.  If we pass a specific session name to the filter, and the service is able to lock that session, it will return a link with the same name.